api/src/main/java/org/openmrs/module/appointmentscheduling/api/AppointmentService.java

```diff
@@ -980,4 +981,40 @@ Integer getHistoryCountByConditions(Date fromDate, Date endDate,
     @Authorized(AppointmentUtils.PRIV_SCHEDULE_APPOINTMENTS)
 	Appointment bookAppointment(Appointment appointment, Boolean allowOverbook)
 			throws TimeSlotFullException;
+
+	/**
+	 * returns daily count for the statuses
+	 * scheduled
+	 * missed
+	 * completed
+	 */
+	@Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+    List<AppointmentDailyCount> getAppointmentsDailyCount(Date fromDate,
+														  Date toDate, Location location, Provider provider, AppointmentType appointmentType,
+														  AppointmentStatus status, VisitType visitType);
+	/**
+	 * returns list of early appointment visits
+	 * who came late/early for their visits
+	 */
+	@Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+	List<Appointment> getEarlyVisits(Date fromDate,
```

Comment:
I would expect a method called "getEarlyVisits" to return a List<Visit>.  You probably want to call this getEarlyAppointments.